### ISSUE TITLE
Add support for integration with TeamCity

### ISSUE THREAD

**Issue by NikolayPianikov** [nunit/nunit#609]
TeamCity uses special launchers for each version of nunit to receive all output in real-time. We would like add supporting for "TEAMCITY_PROJECT_NAME" environment variable to switch on mode of compatibility with TeamCity via TeamCity service messages. It provides ability of integration from the box. When TeamCity runs new process for any build tool it defines process environment variable "TEAMCITY_PROJECT_NAME" which contains the name of build project. Thus user will have good CI even in the case where he configured some complex build steps where nunit was runned indirectly, for example from command line. This variable is specially designed to integrate TeamCity with any build tools. All JetBrains's command tools use this environment variable to start the interaction with TeamCity. Also xUnit and others use it too. (partially done)

**Comment by CharliePoole** [comment]
Thanks for the issue. It makes sense now. I added a reference here from the PR.

I also sent you an invitation to join the contributors group, which will allow me to assign the issue to you in github.

**Comment by CharliePoole** [comment]
Essentially, I'm treating this as only item 2 for the moment. We already support the --teamcity option, as you mention, and we seem to be running parallel tests OK with the merge of PR #590. I'm not sure what Item 4 refers to.

**Comment by NikolayPianikov** [comment]
OK. I will add new issue for item 3 and will remove others except 2

**Comment by CharliePoole** [comment]
That's fine if you see additional issues not resolved in PR #590. Make sure you have that commit in your own repo.

**Comment by NikolayPianikov** [comment]
Ok I will review 590

**Comment by CharliePoole** [comment]
It could be that I have confused the item numbers. :-( Anyway #590 seems to be handling parallel execution in the same process OK but @chris-smith-zocdoc had to remove the suite start and end messages to make it work. When we run in parallel, all the tests run between a suite start and end don't necessarily belong to that suite. I'm not sure how that impacts reporting.
